### ISSUE TITLE
environment.sessionVariables conflict prevents nixos-rebuild

### ISSUE THREAD

**Issue by bjornfor** [NixOS/nixpkgs#20225]
## Issue description

On latest master (aa094fe084ea03146db30117a8a029269463bb86):

```
$ sudo nixos-rebuild boot -I nixpkgs=.
building Nix...
building the system configuration...
error: The unique option `environment.sessionVariables.LD_LIBRARY_PATH' is defined multiple times, in `/home/bfo/proj/code/forks/nixpkgs2/nixos/modules/services/hardware/sane.nix' and `/home/bfo/proj/code/forks/nixpkgs2/nixos/modules/hardware/opengl.nix'.
[fatal error]
```

Both definitions look something like ```environment.sessionVariables.LD_LIBRARY_PATH = [ "foo" ]```.

**Comment by bjornfor** [comment]
Git bisect points to 1fe1cdecb24de63be577a2df538da03bbe96cd1d. @ericsagnes: Do you know how to fix it?

**Comment by ericsagnes** [comment]
I forgot that `either`s didn't merge well, sorry for the trouble.

#20228 should improve the situation, but it will not work for cases where definitions are using different types.

Example, this will merge:

```
environment.sessionVariables.LD_LIBRARY_PATH = [ "foo" ];
environment.sessionVariables.LD_LIBRARY_PATH = [ "bar" ];
```

But this will not:

```
environment.sessionVariables.LD_LIBRARY_PATH = [ "foo" ];
environment.sessionVariables.LD_LIBRARY_PATH = "bar";
```

**Comment by bjornfor** [comment]
Fixed with cf2654b2e0dedd31c6ec14458953c85530db8cec.
